core/nginx/config.py

```diff
@@ -34,6 +34,26 @@
     "notls": None
 }[args["TLS_FLAVOR"]]
 
+def format_for_nginx(fullchain, output):
+    """ We may want to strip ISRG Root X1 out """
+    if not os.path.exists(fullchain):
+        return
+    certs = []
+    with open(fullchain, 'r') as pem:
+        cert = ''
+        for line in pem:
+            cert += line
+            if '-----END CERTIFICATE-----' in line:
+                certs += [cert]
+                cert = ''
+    with open(output, 'w') as pem:
+        for cert in certs[:-1] if len(certs)>2 and os.getenv('LETSENCRYPT_SHORTCHAIN', default="False") else certs:
+            pem.write(cert)
```

Comment:
This can be done shorter and the `default="False"` is wrong (the string "False" is actually True here). What about this?

```
def format_for_nginx(fullchain, output):
    """ We may want to strip ISRG Root X1 out """
    if not os.path.exists(fullchain):
        return
    split = '-----END CERTIFICATE-----\n'
    with open(fullchain, 'r') as pem:
        certs = [f'{cert}{split}' for cert in pem.read().split(split) if cert]
    if len(certs)>2 and os.getenv('LETSENCRYPT_SHORTCHAIN'):
        del certs[-1]
    with open(output, 'w') as pem:
        pem.write(''.join(certs))
```

Reply:
That's in f9869b1d